### PR TITLE
fix #941: clean up API

### LINKED ISSUE #941: Improve docs and api names for processing modules and other containers
## 2) Feature Request

"What is a processing module?"
"What is a data interface?"

These are terms that some users at the hackathon were confused and intimidated by. Either we should explain them more clearly and earlier in the tutorials or rename them to something more approachable to neuroscientists. For example, representing processing modules as folders is pretty intuitive and showing a pictoral representation of an `NWBFile` being a container (or folder) with properties and nested containers would be helpful. 

Also I think renaming `nwbfile.module['my_module_name'].add_data_interface(my_elec_series)` to `nwbfile.processing['my_module_name'].add(my_elec_series)` would be a lot easier for new users to understand.

Relatedly, in the documentation, I think showing multiple options for creating and adding a new field to a container is confusing, e.g. `addProcessingModule` and `createProcessingModule`. Same for getting data, e.g. `nwbfile.module[name]` and `nwbfile.getProcessingModule(name)` (it is also confusing that the naming in this example breaks the convention of `nwbfile.field` == `nwbfile.getField`)

### PR DESCRIPTION
**Pull request by bendichter**
* change ProcessingModule.add_data_interface() to .add()
* change ProcessingModule.get_data_interface() to .get()
* change nwbfile.modules to nwbfile.processing

## Motivation

syntax is now: `nwbfile.processing['my_module_name'].add(my_elec_series)`

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

**Comment by bendichter** [comment]
now that I'm thinking about it maybe we should smoothly deprecate the old way of calling things

**Comment by rly** [comment]
> we should smoothly deprecate the old way of calling things

Yes, I was actually going to comment that for #959 as well.

File: docs/gallery/general/file.py [hunk 1]
```diff
@@ -419,7 +419,7 @@
 
 io = NWBHDF5IO('example_file_path.nwb', mode='a')
 nwbfile = io.read()
-position = nwbfile.modules['behavior'].data_interfaces['Position']
+position = nwbfile.processings['behavior'].data_interfaces['Position']
```
Comment (rly) [inline docs/gallery/general/file.py]: this should be `.processing`

File: src/pynwb/file.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ class NWBFile(MultiContainerInterface):
             'get': 'get_stimulus_template'
         },
         {
-            'attr': 'modules',
```
Comment (rly) [inline src/pynwb/file.py:112]: Does `nwb.modules['module_name']` still work?

File: src/pynwb/base.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def add_container(self, **kwargs):
         '''
         container = getargs('container', kwargs)
         warn(PendingDeprecationWarning('add_container will be replaced by add_data_interface'))
```
Comment (rly) [inline src/pynwb/base.py]: Change warning to `add_container will be replaced by add`

Comment (bendichter) [reply src/pynwb/base.py]:
```suggestion
        warn(PendingDeprecationWarning('add_container will be replaced by add'))
```

File: docs/gallery/general/file.py (hunk 1 above)
Comment (bendichter) [reply docs/gallery/general/file.py]:
```suggestion
position = nwbfile.processing['behavior'].data_interfaces['Position']
```

File: src/pynwb/file.py (hunk 2 above)
Comment (bendichter) [reply src/pynwb/file.py:112]: Yes it still works. I just added a test for it

**Review by rly — APPROVED** [review]

**Pull request merged**